lms/djangoapps/certificates/queue.py

```diff
@@ -281,6 +280,7 @@ def add_cert(self, student, course_id, course=None, forced_grade=None, template_
         self.request.session = {}
 
         is_whitelisted = self.whitelist.filter(user=student, course_id=course_id, whitelist=True).exists()
+        from lms.djangoapps.grades.api import CourseGradeFactory
```

Comment:
Why import this here instead of at the top level?  Can you add a comment if there is a good reason?

Reply:
I believe there was a circular dependency issue - and I was trying different things.  I pushed up a commit to see if the root problem was CMS calling into LMS.

Reply:
Here's the[ stack trace I am getting from Jenkins](https://build.testeng.edx.org/job/edx-platform-python-pipeline-pr/4691/).  I would need to dig into the root cause of this issue.  It's unclear (even after consulting with TestEng) on how to reproduce this issue locally on devstack.

```
12:37:40   File "/edx/app/edxapp/edx-platform/lms/djangoapps/certificates/apps.py", line 24, in ready
12:37:40     from . import signals  # pylint: disable=unused-variable
12:37:40   File "/edx/app/edxapp/edx-platform/lms/djangoapps/certificates/signals.py", line 15, in <module>
12:37:40     from lms.djangoapps.certificates.tasks import generate_certificate
12:37:40   File "/edx/app/edxapp/edx-platform/lms/djangoapps/certificates/tasks.py", line 9, in <module>
12:37:40     from .api import generate_user_certificates
12:37:40   File "/edx/app/edxapp/edx-platform/lms/djangoapps/certificates/api.py", line 27, in <module>
12:37:40     from lms.djangoapps.certificates.queue import XQueueCertInterface
12:37:40   File "/edx/app/edxapp/edx-platform/lms/djangoapps/certificates/queue.py", line 23, in <module>
12:37:40     from lms.djangoapps.grades.api import CourseGradeFactory
12:37:40   File "/edx/app/edxapp/edx-platform/lms/djangoapps/grades/api.py", line 12, in <module>
12:37:40     from lms.djangoapps.grades.tasks import compute_all_grades_for_course as task_compute_all_grades_for_course
12:37:40   File "/edx/app/edxapp/edx-platform/lms/djangoapps/grades/tasks.py", line 51, in <module>
12:37:40     @task(base=LoggedPersistOnFailureTask, routing_key=settings.POLICY_CHANGE_GRADES_ROUTING_KEY)
12:37:40   File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/conf/__init__.py", line 57, in __getattr__
12:37:40     val = getattr(self._wrapped, name)
12:37:40 AttributeError: 'Settings' object has no attribute 'POLICY_CHANGE_GRADES_ROUTING_KEY'
```